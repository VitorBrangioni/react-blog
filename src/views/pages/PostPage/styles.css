.one,
.two,
.three {
  background-color: #212121;
  height: 5px;
  width: 100%;
  margin: 6px auto;
}

.menu-toggle {
  width: 40px;
  position: absolute;
  top: 220px;
  right: 370px;
}

.menu-toggle .one {
  transform: rotate(45deg) translate(7px, 7px);
}
.menu-toggle .two {
  opacity: 0;
}
.menu-toggle .three {
  transform: rotate(-45deg) translate(8px, -9px);
}
