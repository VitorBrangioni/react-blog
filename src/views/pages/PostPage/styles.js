import styled from "styled-components";

export const Main = styled.main`
  padding-top: 5%;
  padding-left: 10%;
  padding-right: 10%;
  background: white;
  grid-column-start: 1;
  grid-column-end: 2;
  width: 100%;
  grid-row-start: 2;
  grid-row-end: null;
`;

export const Container = styled.div`
  display: grid;
  grid-template-columns: 3fr 1fr;
  grid-template-rows: 20vh 80vh;
`;

export const Aside = styled.aside`
  background: #871ae5;
  grid-column: 2/3;
  grid-row: 2 / null;
  padding: 60px 30px;
`;

export const H1 = styled.h1`
  color: #212121;
`;

export const BtnClose = styled.div`
  width: 40px;
  position: absolute;
  top: 25%;
  right: 29%;
  cursor: pointer;
`;

export const LineXOne = styled.div`
  transform: rotate(45deg) translate(7px, 7px);
  background-color: #212121;
  height: 5px;
  width: 100%;
  margin: 6px auto;
`;

export const LineXTwo = styled.div`
  transform: rotate(-45deg) translate(1px, -3px);
  background-color: #212121;
  height: 5px;
  width: 100%;
  margin: 6px auto;
`;
