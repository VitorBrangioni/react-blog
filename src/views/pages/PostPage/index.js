import React, { Component } from "react";
import "../BlogPage/styles.css";
import Header from "../../components/Header";
import RecentPosts from "../../components/RecentPosts";
import { Main, Container, Aside, H1 } from "./styles";

export class PostPage extends Component {
  render() {
    const { title, body } = this.props.location.state.post;

    return (
      <Container>
        <Header />
        <Main>
          <H1>{title}</H1>
          <p>{body}</p>
        </Main>
        <Aside>
          <RecentPosts />
        </Aside>
      </Container>
    );
  }
}

export default PostPage;
