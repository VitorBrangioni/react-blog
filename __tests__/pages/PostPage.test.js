import React from "react";
import PostPage from "../../src/views/pages/PostPage";
import { shallow } from "enzyme";
import posts from "../../__mocks__/posts.mock";

const mockHistoryPush = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe("PostPage", () => {
  it("Snapshot test", () => {
    const wrapper = shallow(
      <PostPage location={{ state: { post: posts[0] } }} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
