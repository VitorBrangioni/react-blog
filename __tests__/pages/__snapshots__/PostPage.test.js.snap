// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPage Snapshot test 1`] = `
<styled.div>
  <HeaderComponent />
  <styled.main>
    <styled.div
      onClick={[Function]}
    >
      <styled.div />
      <styled.div />
    </styled.div>
    <styled.h1>
      AssCo has revamped the theory of versioning
    </styled.h1>
    <styled.p>
      We will mesh the buzzword "strategic". Our feature set is unparalleled, but our subscriber-defined CAE and non-complex configuration is invariably considered a remarkable achievement. What does the commonly-used commonly-used term "strategic" really mean? Think ultra-long-term. The reporting factor can be summed up in one word: B2B2C. Think clicks-and-mortar. We believe we know that it is better to enhance compellingly than to monetize dynamically. Think real-time, backward-compatible. The ability to synergize macro-holistically leads to the capability to envisioneer holistically. The implementation factor can be summed up in one word: six-sigma.
    </styled.p>
    <styled.p>
      Author: 
      Zinnia Rickard
    </styled.p>
    <styled.p>
      Data: 
      2017-04-12
    </styled.p>
  </styled.main>
  <styled.aside>
    <Connect(RecentPosts) />
  </styled.aside>
</styled.div>
`;
